mm/kfence.c

```diff
@@ -85,11 +85,12 @@ static inline void kfence_disable(void)
 	WRITE_ONCE(kfence_enabled, false);
 }
 
-#define KFENCE_WARN_ON(cond) ({	\
-	bool __cond = WARN_ON(cond);	\
-	if (__cond) \
-	kfence_disable(); \
-	__cond; \
+#define KFENCE_WARN_ON(cond)                                                   \
+	({                                                                     \
+		bool __cond = WARN_ON(cond);                                   \
+		if (__cond)                                                    \
```

Comment:
No idea of the compiler is clever enough to have the 'unlikely' in WARN_ON propagates here. However, if this is only used in slow-path, no action required.

Reply:
I think we can just add the 'unlikely' to be on the safe side.